cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/ByteBufferUtils.java

```diff
@@ -222,4 +215,55 @@ public static void skipFully(InputStream is, long length) throws IOException
             throw new EOFException("EOF after " + skipped + " bytes out of " + length);
         }
     }
+
+    // Extract component position from buffer; return null if there are not enough components
+    public static ByteBuffer extractComponent(ByteBuffer buffer, int position)
+    {
+        buffer = buffer.duplicate();
+        readStatic(buffer);
+        int index = 0;
+        while (buffer.remaining() > 0)
+        {
+            ByteBuffer c = readBytesWithShortLength(buffer);
+            if (index == position)
+            {
+                return c;
+            }
+
+            buffer.get();  // Skip end-of-component
+            ++index;
+        }
+        return null;
+    }
+
+    public static ByteBuffer[] split(ByteBuffer name, int numKeys)
+    {
+        // Assume all components, we'll trunk the array afterwards if need be, but most names will be complete
+        ByteBuffer[] l = new ByteBuffer[numKeys];
```

Comment:
Nit: I would avoid single-symbol variable names like `l` (especially `l`, as it looks awfully like `1`).